ingest/ledger_transaction.go

```diff
@@ -181,3 +188,413 @@ func (t *LedgerTransaction) operationChanges(ops []xdr.OperationMeta, index uint
+	if minSequenceLedgerGap == nil {
+		return 0, false
+	}
+
+	return int64(*minSequenceLedgerGap), true
+}
+
+func (t *LedgerTransaction) GetSorobanData() (xdr.SorobanTransactionData, bool) {
+	switch t.Envelope.Type {
+	case xdr.EnvelopeTypeEnvelopeTypeTx:
+		return t.Envelope.V1.Tx.Ext.GetSorobanData()
+	case xdr.EnvelopeTypeEnvelopeTypeTxFeeBump:
+		return t.Envelope.FeeBump.Tx.InnerTx.V1.Tx.Ext.GetSorobanData()
+	default:
+		return xdr.SorobanTransactionData{}, false
+	}
+}
+
+func (t *LedgerTransaction) SorobanResourceFee() (int64, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return int64(sorobanData.ResourceFee), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesInstructions() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.Instructions), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesReadBytes() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.ReadBytes), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesWriteBytes() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.WriteBytes), true
+}
+
+func (t *LedgerTransaction) SorobanInclusionFeeBid() (int64, bool) {
+	resourceFee, ok := t.SorobanResourceFee()
+	if !ok {
+		return 0, false
+	}
+
+	return int64(t.Envelope.Fee()) - resourceFee, true
+}
+
+func (t *LedgerTransaction) FeeAccountAddress() (string, bool) {
+	switch t.Envelope.Type {
+	case xdr.EnvelopeTypeEnvelopeTypeTx:
+		sourceAccount := t.Envelope.SourceAccount()
+		feeAccountAddress := sourceAccount.Address()
+		return feeAccountAddress, true
+	case xdr.EnvelopeTypeEnvelopeTypeTxFeeBump:
+		feeBumpAccount := t.Envelope.FeeBumpAccount()
+		feeAccountAddress := feeBumpAccount.Address()
+		return feeAccountAddress, true
+	default:
+		return "", false
+	}
+}
+
+func (t *LedgerTransaction) SorobanInclusionFeeCharged() (int64, bool) {
+	resourceFee, ok := t.SorobanResourceFee()
+	if !ok {
+		return 0, false
+	}
+
+	feeAccountAddress, ok := t.FeeAccountAddress()
+	if !ok {
+		return 0, false
+	}
+
+	accountBalanceStart, accountBalanceEnd := getAccountBalanceFromLedgerEntryChanges(t.FeeChanges, feeAccountAddress)
+	initialFeeCharged := accountBalanceStart - accountBalanceEnd
+
+	return initialFeeCharged - resourceFee, true
+}
+
+func (t *LedgerTransaction) InclusionFeeCharged() (int64, bool) {
+	inclusionFee, ok := t.SorobanInclusionFeeCharged()
+	if ok {
+		return inclusionFee, ok
+	}
+
+	// Inclusion fee can be calculated from the equation:
+	// Fee charged = inclusion fee * (operation count + if(feeBumpAccount is not null, 1, 0))
+	// or inclusionFee = feeCharged/(operation count + if(feeBumpAccount is not null, 1, 0))
+	operationCount := t.OperationCount()
+	if t.Envelope.Type == 5 { // xdr.EnvelopeTypeEnvelopeTypeTxFeeBump
```

Comment:
```suggestion
	if t.Envelope.Type == xdr.EnvelopeTypeEnvelopeTypeTxFeeBump { 
```
Why not compare against xdr.EnvelopeTypeEnvelopeTypeTxFeeBump instead of 5 ?